python/ray/autoscaler/updater.py

```diff
@@ -271,9 +271,9 @@ def ssh_cmd(self,
             ssh.append("-tt")
         if emulate_interactive:
             force_interactive = (
-                "set -i || true && source ~/.bashrc && "
+                "true && source ~/.bashrc && "
                 "export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && ")
-            cmd = "bash --login -c {}".format(quote(force_interactive + cmd))
```

Comment:
hm I remember there was a reason why we didn't do this...

https://github.com/ray-project/ray/issues/1444

Reply:
The line 276 with `bash -i` is also a way to force the interactive shell, according to `man bash`:

...
OPTIONS
       All  of  the  single-character  shell  options  documented in the description of the set builtin command can be used as options when the shell is
       invoked.  In addition, bash interprets the following options when it is invoked:

       -c        If the -c option is present, then commands are read from the first non-option argument command_string.  If there  are  arguments  after
                 the  command_string,  the  first argument is assigned to $0 and any remaining arguments are assigned to the positional parameters.  The
                 assignment to $0 sets the name of the shell, which is used in warning and error messages.
       -i        If the -i option is present, the shell is interactive.
...

So basically the `set -i` is moved to the options of invoking bash, which is compatible with both bash 4.3 and 4.4.

Maybe this is not a valid test, but I've tried with according to the [answer on ask ubuntu](https://askubuntu.com/questions/879364/differentiate-interactive-login-and-non-interactive-non-login-shell).

Both in ubuntu 16.04 and 18.04:

```bash
bash --login -c 'echo $-;'
#result is hBc, the set status in shell
```

```bash
bash --login -c -i 'echo $-;'
#result is himBHc
```

in Ubuntu 16.04:
```bash
bash --login -c 'set -i || true &&  echo $-;'
#result is hiBc
```

the status i is set after added the `-i` by invoking bash, the side effect is that `m` and `H`  are set either. Checking the meaning of them in the help function of `set`, which gives out:

-m  Job control is enabled.
-H  Enable ! style history substitution.  This flag is on
          by default when the shell is interactive.

So I think these two side effects are not so critical for our use case?

Reply:
I see; OK at a high level it looks OK. 

Before merging I'd (or someone else can do it) want to check whether our default autoscaler scripts work still for AWS and GCP; namely if appending the conda path to the bashrc still has the same behavior as before.

Reply:
I tried this and it seems fine on AWS / DL AMIs.

Reply:
Also seems fine on GCP with the nvidia-docker examples.